### PR TITLE
Post belongs to user

### PR DESCRIPTION
**Pull request by JayWebDevCom**
Posts have a database link to the User that posted them AND posts show the name of the user that posted them.
Unit tests passing.
Feature tests included in Timeline Spec.

Comment tests and posts tests adjusted to include a corresponding user_id.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -40,7 +40,9 @@
     t.datetime "updated_at", null: false
     t.bigint "like_id"
     t.string "image"
+    t.bigint "user_id"
     t.index ["like_id"], name: "index_posts_on_like_id"
+    t.index ["user_id"], name: "index_posts_on_user_id"
   end
```
Comment (adamjohnsnow) [inline db/schema.rb:46]: Are both t.bigint "user_id" and t.index ["user_id"] needed? I don't know the answer, but feels like they do the same job.

File: spec/features/timeline_spec.rb [hunk 2]
```diff
@@ -37,3 +37,16 @@
...
+
+RSpec.feature 'Users can post messages and photos to the timeline', type: :feature do
+  scenario 'Posts show the name of posting user' do
+    sign_in
+    user = User.first
+    click_link 'New post'
+    attach_file('post_image', Rails.root + 'spec/support/assets/owl.jpg')
+    fill_in 'Message', with: 'Test post'
+    click_button 'Submit'
+    expect(page).to have_content "By: #{user.name}"
```
Comment (adamjohnsnow) [inline spec/features/timeline_spec.rb:49]: NICE!

**Review by adamjohnsnow — APPROVED** [review]
Over to @neoeno

File: db/schema.rb (hunk 1 above)
Comment (JayWebDevCom) [reply db/schema.rb:46]: I think so good question. Column `user_id` needs to exits in the `posts` table in order for the index between `posts` and `users` to be created through it.

Comment (neoeno) [reply db/schema.rb:46]: They are! Indexes are different to columns. Worth researching for a spreadsheet aficionado such as yourself @adamjohnsnow!

File: spec/features/timeline_spec.rb [hunk 3]
```diff
@@ -37,3 +37,16 @@
...
+    expect(page).to have_xpath("//img[contains(@src,'owl.jpg')]")
```
Comment (neoeno) [inline spec/features/timeline_spec.rb]: We can probably ditch the owl expect — that way if this test fails we'll know it's nothing to do with the photos.

File: spec/models/comment_spec.rb [hunk 4]
```diff
@@ -1,9 +1,11 @@
 require 'rails_helper'
 
 RSpec.describe Comment, type: :model do
+
+  let(:user) { User.create(name: "Tester", email: "test@example.com", password: "secret") }
+  let(:post) { Post.create(message: "post message", user_id: user.id) }
```
Comment (neoeno) [inline spec/models/comment_spec.rb:6]: Nice refactor here

File: spec/models/post_spec.rb [hunk 5]
```diff
@@ -1,5 +1,24 @@
...
+    post = user.posts.create(message: "this is the post message")
+    expect(post).to be_valid
+  end
+
+  it 'belongs to a user' do
+    user = User.create(name: "User Name", email: "test@example.com",
+    password: "my_secret")
+    post = Post.create(message: 'post message', user_id: user.id)
+    expect(post.user).to eq user
+  end
```
Comment (neoeno) [inline spec/models/post_spec.rb:22]: Great

**Review by neoeno — APPROVED** [review]
Looks good — minor tweak to the feature test and then go ahead and merge away.

**Pull request merged**